frontend/src/features/entity_selector/EntityTextInput.tsx

```diff
@@ -1,50 +1,275 @@
-              <Link />
-            </InputAdornment>
-          ),
-        }}
-        sx={{
-          bgcolor: 'white',
-        }}
-        data-testid="paste-video-url"
-      />
+      <Box p={1}>
+        <Typography
+          variant="subtitle1"
+          textAlign="center"
+          sx={{
+            '& strong': {
+              color: 'secondary.main',
+              fontSize: '1.4rem',
+            },
+          }}
+        >
+          {error ? (
+            t('entitySelector.errorOnLoading')
+          ) : (
+            <Trans
+              t={t}
+              i18nKey="entitySearchInput.searchResults"
+              count={nResults}
+            >
+              <strong>{{ nResults }}</strong> results for {{ lastSearch }}
+            </Trans>
+          )}
+        </Typography>
+      </Box>
+      <EntitySearchResultsList entities={entities} onSelect={onSelect} />
+    </Paper>
+  );
+};
+
+const EntitySearchInput = ({
+  onClose,
+  onResults,
+  onResultSelect,
+}: EntitySearchInputProps) => {
+  const { t } = useTranslation();
+  const theme = useTheme();
+  const { name: pollName } = useCurrentPoll();
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const [search, setSearch] = useState('');
+  const [lastSearch, setLastSearch] = useState('');
+  const [entities, setEntities] = useState<Array<EntityResult>>([]);
+
+  const closeSearch = () => {
+    setSearch('');
+    setLastSearch('');
+    setEntities([]);
+    setError(false);
+    setLoading(false);
+    onClose && onClose();
+  };
+
+  const selectResult = (uid: string) => {
+    closeSearch();
+    onResultSelect && onResultSelect(uid);
+  };
+
+  const searchEntity = async (event: React.FormEvent) => {
+    event.preventDefault();
+    (document.activeElement as HTMLElement).blur();
+
+    if (!search) {
+      return;
+    }
+
+    setLoading(true);
+    const target = extractVideoId(search, { ignoreVideoId: true });
+
+    if (target) {
+      selectResult(target);
+      return;
+    }
+
+    let entities;
+    try {
+      entities = await PollsService.pollsRecommendationsList({
+        name: pollName,
+        search: search,
+        unsafe: true,
+        limit: 30,
+      });
+    } catch (err) {
+      setError(true);
+      setLoading(false);
+      setEntities([]);
+      setLastSearch(search);
+      console.error(err);
+      return;
+    }
+
+    const results = entities.results ?? [];
+
+    setError(false);
+    setLoading(false);
+    setEntities(results);
+    setLastSearch(search);
+    onResults && onResults();
+  };
+
+  const displayResults = () => {
+    if (entities.length > 0) {
+      return true;
+    }
+
+    if (lastSearch !== '') {
+      return true;
+    }
+
+    return false;
+  };
+
+  return (
+    <Box pt={2} bgcolor="grey.100">
+      <form onSubmit={searchEntity} name="entity-selector-search">
+        <Box p={1} display="flex">
```

Comment:
@amatissart is there still a horizontal scrollbar on iOS?

Should we reduce the size of this `Box` with something like `width="98%"` and `margin="auto"`? Maybe using `overflow="hidden"`?